<launch>
    <!--  ################# Input arguments  #################  -->
    <arg name="enableImu" default="false"/>
    <arg name="enableSafety" default="true"/>
    <arg name="autosave_params_path" default="$(find cola2_hug)/config/saved_configurations"/>

    <!--  #################  Robot description  #################  -->
    <node name="robot_state_publisher_hug" pkg="robot_state_publisher" type="state_publisher" >
        <remap from="robot_description" to="robot_description_hug"/>
    </node>

    <!--  #################  Load safety nodes  #################  -->
    <node name="param_logger" pkg="cola2_safety" type="param_logger.py" output="screen" args="$(arg autosave_params_path)"/>
    <node name="up_time" pkg="cola2_safety" type="up_time.py" output="screen"/>
    <node name="hug_mon_control_board" pkg="hug_mon_control_board" type="hug_mon_control_board.py" respawn="true" output="screen"/>
    <node name="safe_depth_altitude" pkg="cola2_safety" type="safe_depth_altitude.py" respawn="true" output="screen"/>
    <node name="emus_bms_client" pkg="emus_bms" type="emus_bms_client" respawn="true" output="screen"/>
    <node name="set_zero_velocity" pkg="cola2_safety" type="set_zero_velocity.py" respawn="true" output="screen"/>
    <node name="set_pose" pkg="cola2_safety" type="set_pose.py"/>
    <node name="recovery_actions" pkg="cola2_safety" type="recovery_actions.py" respawn="true" output="screen"/>
    <node name="virtual_cage" pkg="cola2_safety" type="virtual_cage.py" output="screen"/>
    <node pkg="diagnostic_aggregator" type="aggregator_node" name="diagnostic_aggregator"/>
    <node name="vehicle_status_parser" pkg="cola2_hug" type="vehicle_status_parser.py" output="screen"/>
    <node if="$(arg enableSafety)" name="safety_supervisor" pkg="cola2_safety" type="safety_supervisor.py"/>
    <node name="check_sensors" pkg="cola2_safety" type="check_sensors.py" />
    <node name="computer_logger" pkg="cola2_safety" type="computer_logger.py" respawn="true" output="screen"/>

    <!--  #################  Run Navigation Stack  #################  -->
    <node name="teledyne_explorer_dvl" pkg="teledyne_explorer_dvl" type="teledyne_explorer_dvl" respawn="true"/>
    <group if="$(arg enableImu)">
        <node name="adis_imu" pkg="adis_imu" type="adis_imu" respawn="true" output="screen"/>
        <node name="imu_angle_estimator" pkg="adis_imu" type="imu_angle_estimator.py" respawn="true" output="screen"/>
    </group>
    <node name="fastrax_it_500_gps" pkg="fastrax_it_500_gps" type="fastrax_it_500_gps" respawn="true" output="screen"/>
    <node name="hug_actuators" pkg="hug_actuators" type="hug_actuators" respawn="true"/>
    <node name="navigator" pkg="cola2_hug" type="navigator_hug" respawn="false" output="screen">
        <remap from="/airmar_echosounder" to="/echosounders/e15"/>
    </node>

    <!--  #################  Run Control Stack  #################  -->
    <node name="teleoperation" pkg="cola2_control" type="teleoperation.py" respawn="true" output="screen"/>
    <node name="controller" pkg="cola2_hug" type="controller_hug" respawn="true"/>
    <node name="captain" pkg="cola2_control" type="captain" respawn="false" output="screen"/>
	<node name="pilot" pkg="cola2_control" type="pilot" respawn="false" output="screen"/>

    <!--  #################  Other nodes  #################  -->
    <node name="bag_node" pkg="cola2_lib" type="bag_node.py" output="screen"/>
    <node name="test_thrusters" pkg="cola2_safety" type="test_thrusters.py" output="screen"/>

	</launch>
